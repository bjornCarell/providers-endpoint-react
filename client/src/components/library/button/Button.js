import React from 'react';
import PropTypes from 'prop-types';
import { ButtonStyled } from './ButtonStyled';
import { Loader } from '../loader/Loader';

export const Button = ({
  secondary,
  large,
  inverse,
  loading,
  children,
  ...props
}) => (
  <ButtonStyled
    secondary={secondary}
    large={large}
    inverse={inverse}
    {...props}
  >
    {loading ? <Loader small white /> : children}
  </ButtonStyled>
);

Button.propTypes = {
  secondary: PropTypes.bool,
  large: PropTypes.bool,
  inverse: PropTypes.bool,
  loading: PropTypes.node,
  children: PropTypes.string,
  props: PropTypes.object
};
