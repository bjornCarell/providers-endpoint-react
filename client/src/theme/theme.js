export const theme = {
  color: {
    primary: '#004146',
    primaryLight: '#427783',
    primaryBright: '#E8F9F8',
    secondary: '#F89572',
    secondaryLight: '#FDDFD4',
    secondaryDark: '#FF8A61',
    secondaryBright: '#FFE8E5',
    white: '#FFFFFF',
    light: '#fcfcfc',
    black: '#262626',
    grey: '#F5F6F7'
  },
  spacing: {
    xss: '0.5rem',
    xs: '0.75rem',
    small: '1rem',
    medium: '2rem',
    large: '4rem',
    xl: '8rem',
    xxl: '16rem'
  },
  fontSize: {
    small: '1.2rem',
    medium: '1.6rem',
    large: '2.6rem',
    xl: '4.22rem',
    xxl: '6.85rem'
  },
  fontWeight: {
    small: 200,
    medium: 400,
    large: 600,
    xl: 800
  },
  animation: {
    shadowAround: '0px 0px 10px 0px rgba(0,0,0,0.1)',
    shadowAroundHover: '0px 0px 16px 0px rgba(0,0,0,0.13)',
    shadowBottom: '0px 4px 8px 0 rgba(0,0,0,0.1)',
    shadowBottomHover: '0px 8px 8px 0 rgba(0,0,0,0.12)',
    radiusS: '0.375rem',
    radiusM: '0.875rem',
    radiusL: '2rem',
    transition: 'all .25s cubic-bezier(0.4,0,0.2,1)'
  },
  zIndex: {
    sidebar: 4
  }
};
