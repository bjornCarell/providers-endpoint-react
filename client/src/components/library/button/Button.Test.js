import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import { Button } from './Button';
import { theme } from '../../../theme/theme';

it('renders button without crashing', () => {
  const div = document.createElement('div');
  ReactDom.render(
    <ThemeProvider theme={theme}>
      <Button />
    </ThemeProvider>,
    div
  );
});

it('renders inverse button without crashing', () => {
  const div = document.createElement('div');
  ReactDom.render(<Button theme={theme} />, div);
});

// SNAPSHOTS
it('renders default button correctly', () => {
  const tree = renderer.create(<Button theme={theme}>Tink</Button>).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule(
    'background-color',
    `${theme.color.primaryLight}`
  );
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('color', `${theme.color.white}`);
});

it('renders inverse button correctly', () => {
  const tree = renderer
    .create(
      <Button theme={theme} inverse>
        Think
      </Button>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('background-color', `${theme.color.white}`);
  expect(tree).toHaveStyleRule('color', `${theme.color.primaryLight}`);
});

it('renders round secondary button with box-shadow', () => {
  const tree = renderer
    .create(
      <Button theme={theme} secondary round shadowAround>
        Think
      </Button>
    )
    .toJSON();
  expect(tree).toHaveStyleRule('background-color', `${theme.color.secondary}`);
  expect(tree).toHaveStyleRule('border-radius', `${theme.animation.radiusL}`);
  expect(tree).toHaveStyleRule('box-shadow', `${theme.animation.shadowAround}`);
});
