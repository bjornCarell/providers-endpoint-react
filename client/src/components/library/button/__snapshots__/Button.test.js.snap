// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default button correctly 1`] = `
.c0 {
  background-color: #427783;
  border: 1px solid transparent;
  border-radius: 0.875rem;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  color: #FFFFFF;
  cursor: pointer;
  font-size: 1.2rem;
  font-weight: 600;
  height: auto;
  padding: 1.2rem 2.4rem;
  -webkit-transition: all .25s cubic-bezier(0.4,0,0.2,1);
  transition: all .25s cubic-bezier(0.4,0,0.2,1);
  width: auto;
}

.c0:hover {
  background-color: #004146;
  box-shadow: 0px 8px 8px 0 rgba(0,0,0,0.12);
}

<button
  className="c0"
  height="auto"
  width="auto"
>
  Tink
</button>
`;

exports[`renders default button correctly 2`] = `
<button
  className="sc-bdfBwQ XBVmM"
  height="auto"
  width="auto"
>
  Tink
</button>
`;

exports[`renders inverse button correctly 1`] = `
.c0 {
  background-color: #427783;
  border: 1px solid transparent;
  border-radius: 0.875rem;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  color: #FFFFFF;
  cursor: pointer;
  font-size: 1.2rem;
  font-weight: 600;
  height: auto;
  padding: 1.2rem 2.4rem;
  -webkit-transition: all .25s cubic-bezier(0.4,0,0.2,1);
  transition: all .25s cubic-bezier(0.4,0,0.2,1);
  width: auto;
  background-color: #FFFFFF;
  color: #427783;
}

.c0:hover {
  background-color: #004146;
  box-shadow: 0px 8px 8px 0 rgba(0,0,0,0.12);
}

.c0:hover {
  background-color: #004146;
  color: #FFFFFF;
}

<button
  className="c0"
  height="auto"
  width="auto"
>
  Think
</button>
`;
